compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -96,18 +126,140 @@ object Interactive {
   }
 
+  /** Get possible completions from tree at `pos`
+   *
+   *  @return offset and list of symbols for possible completions
+   */
+  def completions(pos: SourcePosition)(implicit ctx: Context): (Int, List[Symbol]) = {
+    val path = pathTo(ctx.compilationUnit.tpdTree, pos.pos)
+    computeCompletions(pos, path)(contextOfPath(path))
+  }
+
+  private def computeCompletions(pos: SourcePosition, path: List[Tree])(implicit ctx: Context): (Int, List[Symbol]) = {
+    val completions = Scopes.newScope.openForMutations
+
+    val (completionPos, prefix, termOnly, typeOnly) = path match {
+      case (ref: RefTree) :: _ =>
+        if (ref.name == nme.ERROR)
+          (ref.pos.point, "", false, false)
+        else
+          (ref.pos.point,
+           ref.name.toString.take(pos.pos.point - ref.pos.point),
+           ref.name.isTermName,
+           ref.name.isTypeName)
+      case _ =>
+        println(i"COMPUTE from ${path.headOption}")
+        (0, "", false, false)
+    }
+
+    /** Include in completion sets only symbols that
+     *   - start with given name prefix
+     *   - do not contain '$' except in prefix where it is explicitly written by user
+     *   - have same term/type kind as name prefix given so far
+     */
+    def include(sym: Symbol) =
+      sym.name.startsWith(prefix) &&
+      !sym.name.toString.drop(prefix.length).contains('$') &&
+      (!termOnly || sym.isTerm) &&
+      (!typeOnly || sym.isType)
+
+    def enter(sym: Symbol) =
+      if (include(sym)) completions.enter(sym)
+
+    def add(sym: Symbol) =
+      if (sym.exists && !completions.lookup(sym.name).exists) enter(sym)
+
+    def addMember(site: Type, name: Name) =
+      if (!completions.lookup(name).exists)
+        for (alt <- site.member(name).alternatives) enter(alt.symbol)
+
+    def accessibleMembers(site: Type, superAccess: Boolean = true): Seq[Symbol] = site match {
+      case site: NamedType if site.symbol.is(Package) =>
+        site.decls.toList.filter(include) // Don't look inside package members -- it's too expensive.
+      case _ =>
+        def appendMemberSyms(name: Name, buf: mutable.Buffer[SingleDenotation]): Unit =
+          try buf ++= site.member(name).alternatives
+          catch { case ex: TypeError => }
+        site.memberDenots(takeAllFilter, appendMemberSyms).collect {
+          case mbr if include(mbr.symbol) => mbr.accessibleFrom(site, superAccess).symbol
+          case _ => NoSymbol
+        }.filter(_.exists)
+    }
+
+    def addAccessibleMembers(site: Type, superAccess: Boolean = true): Unit =
+      for (mbr <- accessibleMembers(site)) addMember(site, mbr.name)
+
+    def getImportCompletions(ictx: Context): Unit = {
+      implicit val ctx = ictx
+      val imp = ctx.importInfo
+      if (imp != null) {
+        def addImport(name: TermName) = {
+          addMember(imp.site, name)
+          addMember(imp.site, name.toTypeName)
+        }
+        for (renamed <- imp.reverseMapping.keys) addImport(renamed)
```

Comment:
I don't think this can work since the name `renamed` doesn't exist in `imp.site`

Reply:
Good point. I removed the code and added a FIXME. Will open a separate issue.